inc/class-wpseo-utils.php

```diff
@@ -839,6 +839,25 @@ public static function is_yoast_seo_free_page( $current_page ) {
 	}
 
 	/**
+	 * Check if the current page is a Yoast SEO advanced settings page.
+	 *
+	 * @param string $current_page the current page the user is on.
+	 *
+	 * @return bool
+	 */
+	public static function is_yoast_seo_advanced_settings_page( $current_page ) {
```

Comment:
I would introduce a class to check this and to turn on the possible advanced settings.